src/python/s2_common.i

```diff
@@ -597,3 +723,72 @@ USE_COMPARISON_FOR_LT_AND_GT(S1Angle)
 USE_EQUALS_FN_FOR_EQ_AND_NE(S2Loop)
 USE_EQUALS_FN_FOR_EQ_AND_NE(S2Polygon)
 USE_EQUALS_FN_FOR_EQ_AND_NE(S2Polyline)
+
+// repr() methods for geodetic/cell classes
+
+%extend S2CellId {
+  %pythoncode %{
+    def __repr__(self):
+      return '<S2CellId: %s>' % self.ToToken()
```

Comment:
Should this eval back to the same thing? If not, ToString() is more readable and standard for debugging in c++.